core/merge.js

```diff
@@ -725,6 +750,23 @@ class Merge {
     log.debug({ path: doc.path }, 'deleteFileAsync')
     const file /*: ?Metadata */ = await this.pouch.byIdMaybeAsync(doc._id)
     if (!file) return null
+    if (file.moveFrom) {
+      // The file was moved and we don't want to delete it as we think users
+      // delete "paths".
+      if (side === 'remote') {
+        // The file was completely destroyed on the Cozy so it cannot be
+        // restored fromt the trash and should be re-uploaded.
+        delete file.remote
+        delete file.sides.remote
+        delete file.moveFrom
+        return this.pouch.put(file)
+      } else {
+        log.warn(
+          { path: doc.path, doc, file },
+          'deleteFileAsync should not be called for local side'
```

Comment:
You are sure about about that? I can be wrong, but it looks like https://github.com/cozy-labs/cozy-desktop/blob/master/core/prep.js#L132 could do that.

Reply:
Ah crap, I missed this one. Good catch.